src/test.js

```diff
@@ -733,17 +733,11 @@ function internalStart( test ) {
 }
 
 function numberOfTests( module ) {
-	let count = module.tests.length,
-		modules = [ ...module.childModules ];
 
 	// Do a breadth-first traversal of the child modules
-	while ( modules.length ) {
-		let nextModule =  modules.shift();
-		count += nextModule.tests.length;
-		modules.push( ...nextModule.childModules );
-	}
-
-	return count;
+	return module.childModules.reduce( function( p, childModule ) {
+	    return p + numberOfTests( childModule );
```

Comment:
fyi, I don't want to pull the trigger using arrow fns for now. But it looks [Babel does not mess with it](http://babeljs.io/repl/#?babili=false&evaluate=true&lineWrap=false&presets=es2015%2Creact%2Cstage-2&experimental=false&loose=false&spec=false&code=function%20numberOfTests(%20module%2C%20offset%20%3D%200%20)%20%7B%0A%20%20%20%20return%20module.childModules.reduce(%20(%20p%2C%20childModule%20)%20%3D%3E%20numberOfTests(%20childModule%2C%20p%20)%20)%3B%0A%7D).

I can change it if you prefer.